More improvements to Tileset.md

Eauix opened the pull request:
As I've been creating and experimenting with tilesets for a long while now, I've decided to contribute to the tileset.md page, adding a more detailed explanation on how tile attributes work, which attributes are available and why some attributes are the number they have. I also added a guide on how to create special object tiles like coins, bonus blocks and bricks, as the attributes that made these behavior no longer work.

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Eauix committed:
Merge branch 'SuperTux:master' into master

Eauix committed:
Update Tileset.md

Eauix committed:
Update Tileset.md

Vankata453 commented on Tileset.md:
```diff
@@ -68,16 +76,32 @@ It is also possible to extract parts of bigger images to create tiles:
 
 ---
 
+For a tile that looks one way in editor and other way in game, you can set the images separately:
```
```suggestion
For a tile which looks different in the editor, you can set the images separately:
```

Vankata453 commented on Tileset.md:
````diff
@@ -68,16 +76,32 @@ It is also possible to extract parts of bigger images to create tiles:
+  (editor-images "tiles/[tilegroup]/[tile]-editor.png")
+)
+```
+If you want the tile to be invisible in game, you just remove the "(images)" section
````
```suggestion
If you want the tile to be invisible in-game, remove the `(images)` section.
```

Vankata453 commented on Tileset.md:
````diff
@@ -315,6 +376,116 @@ tilegroup.
+    ")
+)
+```
+Be aware that any object propriety that has quotation marks on it, must get a backslash (\\) before any quotation marks, as they are quotation marks inside other quotation marks.
````
```suggestion
Be aware that any quotation mark in `object-data` must be escaped by adding a backslash (`\`) before it, as it's a quotation mark inside other quotation marks.
```

Vankata453 commented on Tileset.md:
```diff
@@ -315,6 +376,116 @@ tilegroup.
+Be aware that any object propriety that has quotation marks on it, must get a backslash (\\) before any quotation marks, as they are quotation marks inside other quotation marks.
+
+---
+If you find it difficult to write the object proprieties you want to add as a tile, you can place this object in your level, open the level in a text editor and copy the information you'll find about this object, for example, if you want a Mr.Snowball, you get in the level file and search for it, you'll find:
```
```suggestion
If you find it difficult to write the object proprieties you want to add as a tile, you can place the object in your level, open the level in a text editor and copy the properties present for the object. For example, if you add a Mr.Snowball to a level and customize it, in the level file you'll find:
```

Vankata453 commented on Tileset.md:
````diff
@@ -315,6 +376,116 @@ tilegroup.
+      (y 128)
+)
+```
+From this information, you may extract what you want to turn into a tile, you must ignore the x and y positions as they will be defined as where the tile is.
````
```suggestion
From this information, you may extract what you want to turn into a tile. You must ignore the `x` and `y` position properties as they will be defined, depending on where the tile is.
```

Vankata453 commented on Tileset.md:
````diff
@@ -315,6 +376,116 @@ tilegroup.
+)
+```
+---
+Now let's be creative, let's say you want to add a bonus block that is orange and has 2 Mr.Icecubes inside it:
````
```suggestion
Now let's be creative, let's say you want to add a bonus block that is orange and has 2 Mr.Iceblocks inside it:
```

provigz reviewed (commented):
Additionally, maybe we should also add info about the relatively new `deprecated` attribute for `(tiles)`.

Eauix committed:
Update Tileset.md

Co-authored-by: Vankata453 <78196474+Vankata453@users.noreply.github.com>

Eauix committed:
Update Tileset.md

Co-authored-by: Vankata453 <78196474+Vankata453@users.noreply.github.com>

Eauix committed:
Update Tileset.md

Co-authored-by: Vankata453 <78196474+Vankata453@users.noreply.github.com>

Eauix committed:
Update Tileset.md

Co-authored-by: Vankata453 <78196474+Vankata453@users.noreply.github.com>

Eauix committed:
Update Tileset.md

Co-authored-by: Vankata453 <78196474+Vankata453@users.noreply.github.com>

Eauix committed:
Update Tileset.md

Co-authored-by: Vankata453 <78196474+Vankata453@users.noreply.github.com>

Eauix committed:
make numbers more visible

I often confused the "13" with a "19", so I'm changing the numbers to Arial font

Eauix committed:
decrease vertical dimension of the image

Eauix committed:
Add deprecation documentation

Eauix commented:
> Additionally, maybe we should also add info about the relatively new `deprecated` attribute for `(tiles)`.

Done

Vankata453 commented:
By the way, you can use "Add suggestions to batch" from the "Files" tab to not have to commit each suggestion separately.

Vankata453 commented on Tileset.md:
```diff
@@ -310,12 +379,153 @@ tilegroup.
+Deprecated Tiles
+================
+
+When a tile becomes outdated due to it being replaced by a different tile / object / decal - or when it is no longer necessary for level design, this tile is usually made deprecated for the sake of compatibility with older levels that were built with them before. Deprecated tiles can still be loaded by the level and will work as they used to, but they're not available to build with, nor can they  ...
```
```suggestion
When a tile becomes outdated due to it being replaced by a different tile / object / decal, or when it is no longer necessary for level design, it is usually made deprecated for the sake of compatibility with older levels which include it. Deprecated tiles can still be loaded by the level and will work as they used to, but they cannot be added to tilegroups, nor can they be copied when editing the level.
```

Vankata453 commented on images/aatriangletypes2.png:
I believe the text can be improved too, just like the numbers?

Vankata453 commented on Tileset.md:
```diff
@@ -7,12 +7,20 @@ These tilesets are included into the level by a `tiles` and `tilegroup` entry.
+
+<hr>
 
 Introduction
 ============
```
Under the "Introduction" -> "Examples" section, all separators are thicker. I believe they should be thinner as to not be mixed up with the ones separating the main parts of the document (1-6).

Eauix replied:
Turns out there's no way of creating a thin line in MD without adding a title. I tried `<hr style="height:1px;">` but it doesn't seem to support CSS. I also tried to add a title with an invisible character, it made the line but created a really large empty space before the line, so, not sure on what to do about this

Eauix committed:
improve text on the slope image

Eauix committed:
Apply suggestions from code review

Co-authored-by: Vankata453 <78196474+Vankata453@users.noreply.github.com>

Eauix committed:
Update Tileset.md

provigz committed:
Remove unneeded <hr>
